pkg/katatestutils/README.md

```diff
@@ -0,0 +1,175 @@
+# Kata test utilities
+
+* [Test Constraints](#test-constraints)
+    * [Usage](#usage)
+        * [Displaying the TestConstraint](#displaying-the-testconstraint)
+        * [Associating an issue with a constraint](#associating-an-issue-with-a-constraint)
+    * [Examples](#examples)
+        * [Skip tests based on user](#skip-tests-based-on-user)
+        * [Skip tests based on distro](#skip-tests-based-on-distro)
+        * [Skip tests based on kernel version](#skip-tests-based-on-kernel-version)
+    * [Full details](#full-details)
+
+This package provides a small set of test utilities.
+
+## Test Constraints
+
+This package provides helper functions that accept user-specified constraints
+that allow you to skip tests.
+
+### Usage
+
+Create a `TestConstraint` object using the `NewTestConstraint()` constructor.
+This takes a single boolean parameter that specifies if debug output is generated.
+
+In each test that has particular test constraints, call the `NotValid()`
+method on the `TestConstraint` object, passing one or more constraints that
+you want to be valid.
+
+The `NotValid()` function returns `true` if any of the specified constraints
+are not available. This allows for a more natural way to code an arbitrarily
+complex test skip as shown in the following example.
+
+The main object is created in the `init()` function to make it available for
+all tests:
+
+```go
+
+import ktu "katatestutils"
+
+var tc ktu.TestConstraint
+
+func init() {
+    tc = NewTestConstraint(true)
+}
+
+func TestFoo(t *testing.T) {
+
+    // Specify one or more constraint functions. If not satisfied, the test
+    // will be skipped.
+    if tc.NotValid(...) {
+        t.Skip("skipping test")
+    }
+
+    // Test code ...
+}
+```
+
+#### Displaying the TestConstraint
+
+Note that you could add the TestConstraint` object to the `Skip()` call as it
```

Comment:
back tick code block messed up in this line?

Reply:
Fixed.